apps/openmw/mwmechanics/mechanicsmanagerimp.cpp

```diff
@@ -1355,9 +1355,39 @@ namespace MWMechanics
 
     void MechanicsManager::actorKilled(const MWWorld::Ptr &victim, const MWWorld::Ptr &attacker)
     {
-        if (attacker.isEmpty() || attacker != getPlayer())
+        if (attacker.isEmpty())
             return;
 
+        bool attackerIsPlayerFollowerOrSummon = false;
+        MWWorld::Ptr player = getPlayer();
+
+        if (attacker != player) {
+            // Check if attacker is one of the player followers
+            std::list<MWWorld::Ptr> followers = getActorsSidingWith(player);
+            if (std::find(followers.begin(), followers.end(), attacker) != followers.end())
+                attackerIsPlayerFollowerOrSummon = true;
+
+            // If not, check if attacker is one of the player summoned creatures
+            if (!attackerIsPlayerFollowerOrSummon) {
+                try {
+                    CreatureStats &playerStats = player.getClass().getCreatureStats(player);
+                    std::map<CreatureStats::SummonKey, int> summonedCreatures = playerStats.getSummonedCreatureMap();
+
+                    typedef std::map<CreatureStats::SummonKey, int>::iterator it_type;
+                    for (it_type it = summonedCreatures.begin(); it != summonedCreatures.end(); ++it) {
+                        if (attacker.getClass().getCreatureStats(attacker).matchesActorId(it->second)) {
+                            attackerIsPlayerFollowerOrSummon = true;
+                            break;
+                        }
+                    }
+                } catch (std::runtime_error e) {}
```

Comment:
```attacker``` is expected to be an actor which has CreatureStats. If that assumption is violated for some reason, we certainly don't want to ignore the exception, we want to log it and abort whatever caused it. But since we already have plenty of exception catching higher up the stack, there is no need to bother with it here. I suggest to remove the try-catch all together.

This is all just for future reference, since I've already suggested dropping the if statement entirely.